src/transformers/models/llama/modeling_llama.py

```diff
@@ -976,9 +977,11 @@ def forward(
         if use_cache:  # kept for BC (cache positions)
             if not isinstance(past_key_values, StaticCache):
                 past_key_values = DynamicCache.from_legacy_cache(past_key_values)
-            past_seen_tokens = past_key_values.get_seq_length()
+                past_seen_tokens = past_key_values.get_seq_length()
 
         if cache_position is None:
+            if isinstance(past_key_values, StaticCache):
+                raise ValueError("cache_position is a required argument when using StaticCache.")
```

Comment:
```suggestion
        if cache_position is None and isinstance(past_key_values, StaticCache):
                raise ValueError("cache_position is a required argument when using StaticCache.")
```

Reply:
I think the `cache_position = torch.arange` is necessary no (for `cache_position is None`)?

Reply:
sorry yes !